src/hooks.ts

```diff
@@ -1,13 +1,75 @@
+import React from 'react';
 import AsyncStorage from './AsyncStorage';
-import type { AsyncStorageHook } from './types';
 
-export function useAsyncStorage(key: string): AsyncStorageHook {
-  return {
-    getItem: (...args) => AsyncStorage.getItem(key, ...args),
-    setItem: (...args) => AsyncStorage.setItem(key, ...args),
-    mergeItem: (...args) =>
+export function useAsyncStorage(key: string) {
+  const getItem = React.useCallback(
+    (...args) => AsyncStorage.getItem(key, ...args),
+    [key]
+  );
+
+  const setItem = React.useCallback(
+    (...args) => AsyncStorage.setItem(key, ...args),
+    [key]
+  );
+
+  const mergeItem = React.useCallback(
+    (...args) =>
       AsyncStorage.mergeItem?.(key, ...args) ??
       Promise.reject('Not implemented'),
-    removeItem: (...args) => AsyncStorage.removeItem(key, ...args),
+    [key]
+  );
+
+  const removeItem = React.useCallback(
+    (...args) => AsyncStorage.removeItem(key, ...args),
+    [key]
+  );
+
+  return {
+    getItem,
+    setItem,
+    mergeItem,
+    removeItem,
   };
 }
+
+export function useAsyncStorageValue(
```

Comment:
I'd rather avoid adding more hooks into the mix. Fixing `useAsyncStorage` + tests would be sufficient for now 🙏

Reply:
OK. I'll update the PR and remove the code.